utils/git.go

```diff
@@ -143,11 +159,44 @@ func (m *manager) readRevision() {
 		break
 	}
 	if err := scanner.Err(); err != nil {
-		errorutils.CheckError(err)
-		m.err = err
+		m.err = errorutils.CheckError(err)
 		return
 	}
 
 	m.revision = revision
 	return
 }
+
+func (m *manager) readRevisionFromPackedRef(ref string) {
+	packedRefPath := filepath.Join(m.path, "packed-refs")
+	exists, err := fileutils.IsFileExists(packedRefPath, false)
+	if err != nil {
+		m.err = err
+		return
+	}
+	if exists {
+		file, err := os.Open(packedRefPath)
+		if errorutils.CheckError(err) != nil {
+			m.err = err
+			return
+		}
+		defer file.Close()
+
+		scanner := bufio.NewScanner(file)
+		for scanner.Scan() {
+			line := scanner.Text()
+			if strings.HasSuffix(line, ref) {
+				split := strings.Split(line, " ")
+				m.revision = split[0]
```

Comment:
Question - shouldn't we add some validation here? Since I'm not sure what the expected line structure is, I'm not sure abput it. Do we want to raise an error here in case the split returned only one section?
I think it is good to add a comment with the expected line structure.

Reply:
I added a validation to the split which I think is sufficient, and an error raise. I cannot add a validation on the revision's length since it could be abbreviated. What do you think?